contracts/modules/cover/Cover.sol

```diff
@@ -253,38 +237,49 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
       )
     );
 
-
     // Update totalActiveCover
     {
-      ActiveCover storage _activeCover = activeCover[params.coverAsset];
-      uint lastUpdateId = _activeCover.lastBucketUpdateId;
+      ActiveCover memory _activeCover = activeCover[params.coverAsset];
+
       uint currentBucketId = block.timestamp / BUCKET_SIZE;
+      uint totalActiveCover = _activeCover.totalActiveCoverInAsset;
+
+      if (totalActiveCover != 0) {
+        totalActiveCover -= getExpiredCoverAmount(
+          params.coverAsset,
+          _activeCover.lastBucketUpdateId,
+          currentBucketId
+        );
+      }
+
+      totalActiveCover -= previousSegmentAmount;
+      totalActiveCover += coverAmountInCoverAsset;
+
       _activeCover.lastBucketUpdateId = currentBucketId.toUint64();
-      // Get expired cover amount
-      expiredCoverAmount += getExpiredCoverAmount(params.coverAsset, lastUpdateId, currentBucketId);
-      // Update total active cover in storage
-      _activeCover.totalActiveCoverInAsset = (
-        _activeCover.totalActiveCoverInAsset
-        + coverAmountInCoverAsset
-        - expiredCoverAmount
-      ).toUint192();
-
-      // Update amount to expire at the end of this cover segment
+      _activeCover.totalActiveCoverInAsset = totalActiveCover.toUint192();
+
+      // update total active cover in storage
+      activeCover[params.coverAsset] = _activeCover;
+
+      // update amount to expire at the end of this cover segment
       uint bucketAtExpiry = Math.divCeil(block.timestamp + params.period, BUCKET_SIZE);
       activeCoverExpirationBuckets[params.coverAsset][bucketAtExpiry] += coverAmountInCoverAsset;
     }
 
-
-    if (amountDueInNXM > 0) {
-      retrievePayment(
-        amountDueInNXM,
-        params.paymentAsset,
-        params.maxPremiumInAsset,
-        params.commissionRatio,
-        params.commissionDestination
-      );
+    // can pay with cover asset or nxm only
+    if (params.paymentAsset != params.coverAsset && params.paymentAsset != NXM_ASSET_ID) {
```

Comment:
this means that we're no longer allowing you to buy an ETH cover with DAI for example.

Was this product level decision made for a while or are we doing it just to simplify things here?

Reply:
this came as a request from Jack - discussion here: https://nexusmutual.slack.com/archives/G010M0U3BT8/p1674141183072719

Reply:
This has been discussed and that was the request.

Reply:
acknowledged